build.gradle

```diff
@@ -7,38 +7,56 @@
 // PLUGINS
 // -----------------------------------------------------------------------------
 
+buildscript {
+    repositories {
+        maven {
+            url 'https://plugins.gradle.org/m2/'
+        }
+    }
+    dependencies {
+        classpath 'io.github.gradle-nexus:publish-plugin:1.3.0'
+    }
+}
+
 plugins {
     // Apply the java-library plugin to add support for Java Library
     id 'java-library'
 
     // Apply the groovy plugin to also add support for Groovy (needed for Spock)
     id 'groovy'
 
-    // Apply the spotbugs plugin for code quality
-    id 'com.github.spotbugs' version '2.0.0'
+    // Apply the brightSPARK Labs plugin for standardisation.
+    id 'com.brightsparklabs.gradle.baseline' version '4.0.0'
 
-    // Apply the jacoco plugin for test coverage
-    id 'jacoco'
+    // -----------------------------------------------------------------------------
+    // PUBLISHING
+    // -----------------------------------------------------------------------------
 
-    // Apply the annotation processing plugins
-    id "net.ltgt.apt" version "0.21"
-    id "net.ltgt.apt-idea" version "0.21"
+    id 'maven-publish'
+    // Maven Central requires artifacts to be signed.
+    id 'signing'
+}
 
-    // Apply bintray plugin for publishing artifacts
-    id "com.jfrog.bintray" version "1.8.4"
-    id "maven-publish"
+/*
+ * Maven Central requires artifacts to be released via Sonatype Nexus.
+ * The Nexus Publish Plugin can only be applied to the root project.
+ * Do not apply it if this project is being included as a subproject.
+ */
+def isRootProject = project.path == ':'
+if (isRootProject) {
+    apply plugin: 'io.github.gradle-nexus.publish-plugin'
 }
```

Comment:
Ah right, you want to conditionally apply it. Right, I suggest we add a comment above the buildscript block then, since its a deviation from standard practice.

Reply:
Done.